EIPS/eip-3643.md

```diff
@@ -90,19 +86,20 @@ To be able to perform a transfer on T-REX you need to fulfill several conditions
 - The receiver needs to be whitelisted on the Identity Registry and verified (hold the necessary claims on his [ONCHAINID](https://github.com/onchain-id/solidity))
 - The sender's wallet cannot be frozen
 - The receiver's wallet cannot be frozen
+- The token cannot be paused
 - The transfer has to respect all the rules of compliance defined in the `Compliance` smart contract (`canTransfer` needs to return `TRUE`)
 
 Here is an example of `transfer` function implementation : 
 ```solidity
 function transfer(address _to, uint256 _amount) public override whenNotPaused returns (bool) {
-        require(!frozen[_to] && !frozen[msg.sender], 'wallet is frozen');
-        require(_amount <= balanceOf(msg.sender).sub(frozenTokens[msg.sender]), 'Insufficient Balance');
-        if (tokenIdentityRegistry.isVerified(_to) && tokenCompliance.canTransfer(msg.sender, _to, _amount)) {
-            tokenCompliance.transferred(msg.sender, _to, _amount);
+        require(!_frozen[_to] && !_frozen[msg.sender], "wallet is frozen");
+        require(_amount <= balanceOf(msg.sender) - (_frozenTokens[msg.sender]), "Insufficient Balance");
+        if (_tokenIdentityRegistry.isVerified(_to) && _tokenCompliance.canTransfer(msg.sender, _to, _amount)) {
```

Comment:
I suggest simplifying the code by using a require statement instead of an if condition. By making this change, we can remove the revert statement and directly return true. This adjustment ensures that the behavior of the code aligns with the ERC20 standard.

Reply:
To simplify the code and improve readability, I suggest these two options:

Option 1: Use a single **require** statement:
```solidity
require(
    _tokenIdentityRegistry.isVerified(_to) && _tokenCompliance.canTransfer(msg.sender, _to, _amount),
    "Transfer not possible"
);
```

Option 2: Split it into two **require** statements for more precise error messages:

```solidity
require(
    _tokenIdentityRegistry.isVerified(to),
    "ERC-3643: Unverified identity"
);
require(
    _tokenCompliance.canTransfer(from, to, amount),
    "ERC-3643: Compliance failure"
);
```

Reply:
can be more clear for people not knowing anything about the standard to separate in 2 require statements indeed